### PR TITLE
Restore missing code

### PR DESCRIPTION
**Pull request by fabien-zimmer**

### CODE REVIEW

**Review by lana-b12 — COMMENTED** [review]
Thanks for the PR. I have only a small question about why the diabetes case is handled differently. It would be nice to have a reference to a ticket / some documentation

File: x-api/utils/efact-util.ts [hunk 1]
```diff
@@ -313,10 +338,13 @@ function toInvoiceItem(
...
   invoiceItem.codeNomenclature = Number(invoicingCode.tarificationId!!.split("|")[1])
   invoiceItem.dateCode = dateEncode(toMoment(invoicingCode.dateCode!!)!!.toDate())
-  invoiceItem.endDateCode =
-    invoiceItem.codeNomenclature === 109594
-      ? dateEncode(toMoment(invoicingCode.dateCode!!)!!.toDate())
-      : dateEncode(toMoment(invoicingCode.dateCode!!)!!.endOf("month").toDate())
+  // Only applies to flatrate invoicing
+  if(flatrateInvoice) {
+    invoiceItem.endDateCode =
+      invoiceItem.codeNomenclature === 109594 // Diabetes pre-care pathways
```
Comment (lana-b12) [inline x-api/utils/efact-util.ts:344]: Is there some documentation about why the diabetes one is different? Can it (or a reference) be added as a comment?

File: x-api/utils/efact-util.ts [hunk 2]
```diff
@@ -226,9 +226,34 @@ export function toInvoiceBatch(
             )
 
             const now = new Date()
-            const invoiceDate = toMoment(invoicesWithPatient[0].invoice.invoiceDate!!)
-            const invoicingMonth = invoiceDate!!.month() + 1
-            const invoicingYear = invoiceDate!!.year()
+            const invoiceDates = _.groupBy(
+              invoicesWithPatient.map(
+                inv =>
+                  !!inv && !!inv.invoice && !!inv.invoice.invoiceDate
```
[the remark this answers is not in the document]
Comment (fabien-zimmer) [reply x-api/utils/efact-util.ts:232]: Almost: we could eventually end up with "_undefined01_". That being said, line could indeed be simplified. 
How about _(inv?.invoice?.invoiceDate ?? 100001).toString().substring(0, 6) + "01"_ . I believe this to be preferred.

File: x-api/utils/efact-util.ts (hunk 1 above)
Comment (fabien-zimmer) [reply x-api/utils/efact-util.ts:344]: Sure, juste came up with https://medispring.atlassian.net/wiki/spaces/EN/pages/2536013825/Behavior+of+diabetes+pre-care+pathways+flatrate+invoicing

File: x-api/utils/efact-util.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lana-b12) [reply x-api/utils/efact-util.ts:232]: I see. I agree with your proposal :)

File: x-api/utils/efact-util.ts (hunk 1 above)
Comment (lana-b12) [reply x-api/utils/efact-util.ts:344]: Thanks!

**Pull request merged**